src/dnnl/dnnl_graph_compiler.cpp

```diff
@@ -43,54 +41,112 @@
 
 // dnnl_graph_compiler.h interface implementation.
 
+using namespace mlir;
+
+extern "C" {
+extern int gc_runtime_keep_alive;
+}
+
 struct dnnl_graph_compiler_executable {
-  // TODO: Implement
+  dnnl_graph_compiler_executable(MLIRContext &context,
+                                 const std::string_view &json)
+      : inputIds(), outputIds(), jmod() {
+    auto mod = JsonParser::parse(context, json, inputIds, outputIds, strides);
+    auto jmod = gc::JitModule::create(mod);
+
+    if (!static_cast<bool>(jmod)) {
+      auto err = jmod.takeError();
+      llvm::errs() << err;
+      llvm::consumeError(std::move(err));
+      std::string msg("Failed to create JitModule: ");
+      llvm::raw_string_ostream(msg) << err;
+      throw std::runtime_error(msg);
+    }
+
+    this->jmod = *jmod;
+  }
 
   void execute(dnnl_graph_compiler_tensor *inputs,
-               dnnl_graph_compiler_tensor *outputs) const;
+               dnnl_graph_compiler_tensor *outputs) const {
+    std::vector<MemRefWrapper> memRefs;
+    memRefs.reserve(inputIds.size() + outputIds.size());
+    for (auto &pair : {std::make_pair(&inputIds, inputs),
+                       std::make_pair(&outputIds, outputs)}) {
+      auto ids = pair.first;
+      auto tensors = pair.second;
+      for (auto id : *ids) {
```

Comment:
I think it's not necessary to use O(N^2) algorithms here, and since this is execution API, we want to have less check and thus less overhead. please simplify this.

Reply:
The `jmod` will respect the argument order set from `mod`. It means if we use same argument order for kernel compilation, then we don't need to reorder it during kernel execution.

Reply:
We use the same order, but the arguments are passed from oneDNN and I'm not sure if the order is always the same as in mod.

Reply:
Now it's O(N), if the arguments order matches.